environment/k8s_environment.go

```diff
@@ -785,12 +873,41 @@ func (m *K8sManifest) setValues() error {
 	return nil
 }
 
+type TemplateValuesArray struct {
+	Values []interface{}
+}
+
+func (t *TemplateValuesArray) next() (interface{}, error) {
+	if len(t.Values) > 0 {
+		valueToReturn := t.Values[0]
+		t.Values = t.Values[1:]
+		return valueToReturn, nil
+	} else {
+		return nil, errors.New("No more Values in the array")
+	}
+}
+
+var mu sync.Mutex
```

Comment:
This shouldn't be a global, needs to be stored in the instance of `TemplateValuesArray`.